addons/recolor-custom-blocks/userscript.js

```diff
@@ -0,0 +1,573 @@
+      });
+    } else {
+      block.updateColour(this);
+      const isProceduresPrototype = block.type === "procedures_prototype"
+      block.getChildren().forEach((child) => {
+        // Make sure children's stroke color isn't messed up
+        // Also for some reason procedures_prototype's stroke color refuses to update, so we set it the hard way
+        isProceduresPrototype && (child.colourTertiary_ = Blockly.Colours[child.getCategory() ?? "more"].tertiary);
+        child.updateColour();
+      });
+    }
+  }
+
+  const handleBlock = (block) => {
+    // Get type of block
+    const blockStyle = Blockly.registry ? block.getStyleName() : (block.getCategory() ?? "more")
+    // We only want to let procedure_ blocks through, but sometimes procedure's type and category aren't set right
+    // It's easier to just do a blacklist
+    if(blockStyle !== "more" || (block.type ?? "").startsWith("argument")) {
+      return;
+    }
+
+    // There used to be a lot of shenanigans here, but with the applyColour / updateColour
+    // methods, we don't generally deal with blocks that haven't been rendered yet
+    const textContent = block.procCode_ ?? ""
+
+    const firstColonIndex = textContent.indexOf(":");
+    if(firstColonIndex !== -1 && !addon.self.disabled) {
+      const colorCandidateString = textContent.substring(0,firstColonIndex);
+
+      const colorCandidate = getBlocklyColors(colorCandidateString);
+      // If we can get a valid Blockly color from colorCandidateString, set the block color
+      if(colorCandidate) {
+        setBlockColor(block, colorCandidate, true, firstColonIndex);
+        return;
+      }
+      // If we can get a valid Hex color from the colorCandidateString, make fake colors
+      // for it and set the block color
+      if(isHexColor(colorCandidateString)) {
+        const fakeColor = getFakeBlockColors(colorCandidateString);
+        setBlockColor(block, fakeColor, true, firstColonIndex);
+
+        return;
+      }
+    }
+    // If the block doesn't qualify for a color change, check to see if it needs to be reverted
+    if(block.recolorCustomBlock?.isEdited) {
+      setBlockColor(block, getBlocklyColors("more"), false, 0);
+    }
+  }
+
+  const updateDeclarationPrefix = (block, prefix) => {
+    const recolorCustomBlock = block.recolorCustomBlock;
+    if(recolorCustomBlock?.isEdited) {
+      block.procCode_ = block.procCode_.slice(recolorCustomBlock?.prefixEnd);
+    }
+    if(prefix !== "more") {
+      block.procCode_ = prefix + ":" + block.procCode_;
+    }
+
+    block.updateDisplay_();
+    handleBlock(block)
+  }
+
+  const createColorButton = (colors) => {
+
+    const colorButton = document.createElement("button");
+    colorButton.classList.add("sa-rcb-colorButton")
+    colorButton.style.backgroundColor = colors.colourPrimary;
+    colorButton.style.borderColor = colors.colourTertiary;
+
+    const checkmarkSvg = document.createElementNS('http://www.w3.org/2000/svg', 'svg')
```

Comment:
Could this be split off into an svg file?

Reply:
> It's path is styled by scratch's `.blocklyFlyoutCheckboxPath` selector (in case an addon modifies the default checkbox). Can you still style non-inline svg elements with external css?

No, but I don't think it makes sense for other addons to affect these check icons anyway - they're quite different from those in the flyout.